scripts/validation/providers/validate.ts

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable no-console */
+import path from 'node:path';
+import fs from 'node:fs';
+import { fileURLToPath } from 'node:url';
+import jsYaml from 'js-yaml';
+import Ajv from 'ajv';
+import chalk from 'chalk';
+import type { Provider } from '@nangohq/types';
+
+console.log('🕵️‍♂️ Validate providers.yaml');
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const pathSchema = path.join(__dirname, 'schema.json');
+const pathProviders = path.join(__dirname, '../../../packages/shared/providers.yaml');
+
+// Schema
+const ajv = new Ajv({ allErrors: true, discriminator: true });
+const schemaJson = fs.readFileSync(pathSchema);
+const validator = ajv.compile(JSON.parse(schemaJson.toString()));
+console.log('loaded schema.json', pathSchema);
+
+// Providers
+const providersYaml = fs.readFileSync(pathProviders);
+console.log('loaded providers.yaml', pathProviders, providersYaml.toString().length);
+const providersJson = jsYaml.load(providersYaml.toString()) as Record<string, Provider>;
+console.log('parsed providers', Object.keys(providersJson));
+
+// Validation
+console.log('validating...');
+validator(providersJson);
+if (validator.errors) {
+    console.error(chalk.red('error'), validator.errors);
+    process.exit(1);
+}
+
+console.log('✅ JSON schema valid');
+
+// Check if files exist
+console.log('Checking values...');
+const docsPath = path.join(__dirname, '../../../docs-v2/integrations/all');
+const svgPath = path.join(__dirname, '../../../packages/webapp/public/images/template-logos');
+
+let error = false;
+for (const [providerKey, provider] of Object.entries(providersJson)) {
```

Comment:
not a big deal but we could have the logic in this loop extracted to a `validateProvider(providerKey, provider)` function to make it more readable. That could also allow to process all providers in parrallel :)

Reply:
Happy to abstract a bit indeed. I won't do it in parallel it's messing up the logs, it's not too slow rn